circuitpython_build_tools/build.py

```diff
@@ -64,6 +68,35 @@ def version_string(path=None, *, valid_semver=False):
 def mpy_cross(mpy_cross_filename, circuitpython_tag, quiet=False):
     if os.path.isfile(mpy_cross_filename):
         return
+
+    # Try to pull from S3
+    uname = os.uname()
+    s3_url = None
+    if uname[0] == 'Linux' and uname[4] in ('amd64', 'x86_64'):
+        s3_url = f"{S3_MPY_PREFIX}mpy-cross.static-amd64-linux-{circuitpython_tag}"
+    elif uname[0] == 'Linux' and uname[4] == 'armv7l':
+        s3_url = f"{S3_MPY_PREFIX}mpy-cross.static-raspbian-{circuitpython_tag}"
+    elif uname[0] == 'Darwin' and uname[4] == 'x86_64':
+        s3_url = f"{S3_MPY_PREFIX}mpy-cross-macos-catalina-{circuitpython_tag}"
+    else:
+        print(f"\nUnable to check S3 for sysname='{uname[0]}' release='{uname[2]}' machine='{uname[4]}'.")
+        print("  Please file an issue at https://github.com/adafruit/circuitpython-build-tools/ with the above message")
+
+    if s3_url is not None:
+        print(f"Checking S3 for {s3_url}")
+        try:
+            r = requests.get(s3_url)
+            if r.status_code == 200:
+                with open(mpy_cross_filename, "wb") as f:
+                    f.write(r.content)
+                    # Set the User Execute bit
+                    os.chmod(mpy_cross_filename, os.stat(mpy_cross_filename)[0] | stat.S_IXUSR)
```

Comment:
this doesn't set the group/other X bits but it's unlikely to cause problems for many real-world cases.

Reply:
While on Linux my standard command would be `chmod +x filename` for usr/grp/oth I don't think we need anything beyond user in this case. In the Travis/Actions it will be getting erased minutes later. If run locally, I don't really see anyone using the built/downloaded `mpy-cross` at the command line but I can change it if you want.

Reply:
nah, it's probably not important.